src/cluster.c

```diff
@@ -5099,30 +5270,14 @@ void addShardReplyForClusterShards(client *c, clusterNode *node, uint16_t *slot_
  * information about each node. */
 void clusterReplyShards(client *c) {
     void *shard_replylen = addReplyDeferredLen(c);
```

Comment:
Since we moved this to a dictionary, we can set the length up front instead of using a deferred length.
```suggestion
    void *shard_replylen = addReplyArrayLen(c, dictSize(server.cluster->shards));
```